### PR TITLE
tests: benchmarks: multicore: idle_spim_loopback: Add test for SPIM120

### PR DESCRIPTION
**Pull request by nordic-segl**
Add test that combinest functional check (whether SPIM transmits data) with non-functional check (current consumption).

Confirm that slow SPIM130 / fast SPIM120:
- doesn't block entering low power states like IDLE or S2RAM;
- work correclty after leaving low power state.

### CODE REVIEW

File: tests/benchmarks/multicore/idle_spim_loopback/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+common:
+  sysbuild: true
+  tags: ci_build ci_tests_benchmarks_multicore
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_spim_loopback/testcase.yaml:3]: please add tag ppk_power_measure

File: tests/benchmarks/multicore/idle_spim_loopback/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  benchmarks.multicore.idle_spim_loopback.nrf54h20dk_cpuapp_cpurad.idle:
+    tags: ppk_power_measure
+    extra_args:
+      SB_CONF_FILE=sysbuild/nrf54h20dk_nrf54h20_cpurad.conf
+      idle_spim_loopback_CONF_FILE=prj_s2ram.conf
+      idle_spim_loopback_CONFIG_TEST_SLEEP_DURATION_MS=500
+    harness: pytest
+    harness_config:
+      fixture: ppk_power_measure spi_loopback
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_spim_loopback/testcase.yaml:31]: only single fixture is allowed, please keep spi_loopback

File: tests/benchmarks/multicore/idle_spim_loopback/testcase.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  benchmarks.multicore.idle_spim_loopback.nrf54h20dk_cpuapp_cpurad.idle_fast:
+    tags: ppk_power_measure
+    extra_args:
+      SB_CONF_FILE=sysbuild/nrf54h20dk_nrf54h20_cpurad.conf
+      idle_spim_loopback_CONF_FILE=prj_s2ram.conf
+      idle_spim_loopback_CONFIG_TEST_SLEEP_DURATION_MS=500
+      idle_spim_loopback_DTC_OVERLAY_FILE="boards/nrf54h20dk_nrf54h20_cpuapp_fast.overlay"
+    harness: pytest
+    harness_config:
+      fixture: ppk_power_measure
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_spim_loopback/testcase.yaml:69]: spi_loopback

File: tests/benchmarks/multicore/idle_spim_loopback/testcase.yaml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+  benchmarks.multicore.idle_spim_loopback.nrf54h20dk_cpuapp_cpurad.s2ram_fast:
+    tags: ppk_power_measure
+    extra_args:
+      SB_CONF_FILE=sysbuild/nrf54h20dk_nrf54h20_cpurad.conf
+      idle_spim_loopback_CONF_FILE=prj_s2ram.conf
+      idle_spim_loopback_DTC_OVERLAY_FILE="boards/nrf54h20dk_nrf54h20_cpuapp_fast.overlay"
+    harness: pytest
+    harness_config:
+      fixture: ppk_power_measure
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_spim_loopback/testcase.yaml:81]: spi_loopback

File: tests/benchmarks/multicore/idle_spim_loopback/src/main.c [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+#define MEMORY_SECTION(node)                                                           \
+	COND_CODE_1(DT_NODE_HAS_PROP(node, memory_regions),                                \
+		    (__attribute__((__section__(                                               \
+			    LINKER_DT_NODE_REGION_NAME(DT_PHANDLE(node, memory_regions)))))),      \
+		    ())
+
+static uint8_t spim_buffer[32] MEMORY_SECTION(DT_BUS(DT_NODELABEL(dut_spi_dt)));
+
+#define DATA_FIELD_LEN	(4)
```
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_spim_loopback/src/main.c:31]: I see weird behaviour when I change data length to 8 bytes (or more) per single transmission.
Current measurements shows "inactivity" while SPI is extensively exchanging data.
For 16 bytes I sow SPI active for 20 ms every 200ms. Remaining 180ms was like IDLE.

File: tests/benchmarks/multicore/idle_spim_loopback/testcase.yaml [hunk 6]
```diff
@@ -0,0 +1,83 @@
+common:
+  sysbuild: true
+  tags: ci_build ci_tests_benchmarks_multicore spim
```
Comment (nordic-segl) [inline tests/benchmarks/multicore/idle_spim_loopback/testcase.yaml:3]: Test that shall have PPK connected, override this field with `tags: ppk_power_measure`. See line 24, 36, 61, 74.

**Comment by nordic-segl** [comment]
All tests are passing here:
https://jenkins-ncs.nordicsemi.no/job/latest/job/test-fw-twister-ncs-zephyr/job/twister-power-consumption-testing-customizable/job/main/71/testReport/nrf54h20dk_nrf54h20_cpuapp_benchmarks.multicore.idle_spim_loopback.nrf54h20dk_cpuapp_cpurad/

When single transmission exchanges 4 bytes of data:
- IDLE (slow instance)
active/sleep: 1 850 / 22 uA
- IDLE (fast instance)
active/sleep: 2 730 / 22 uA
- S2RAM (slow instance)
active/sleep: 1 858 / 3 uA
- S2RAM (fast instance)
active/sleep: 2 734 / 3 uA

**Review by nordic-piks — APPROVED** [review]

**Review by nordic-pikr — APPROVED** [review]

**Pull request merged**